### DISCUSSION TITLE
[Action Advised] Enable email enumeration protection on all of your Firebase projects

### DISCUSSION THREAD

**Discussion by appsgenie** [invertase/react-native-firebase#7596] [General]
I got this email from Firebase titled `[Action Advised] Enable email enumeration protection on all of your Firebase projects`. Could someone help me understand if this affects the rn firebase package too, if we enable this protection? The email links to [https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection](url). It mentions the native `fetchSignInMethodsForEmail` method; I only use `signInWithEmailAndPassword` and wondering if they are connected?
thank you in advance

**Comment by tharwi** [comment]
I have the same question

**Reply by mikehardy** [reply]
`signInWithEmailAndPassword` will work fine

But if you enable email enumeration protection then `fetchSignInMethodsForEmail` will return an empty array, as they document upstream in the description of how APIs respond differently with protection enabled.

There is nothing more to it than that. If you turn email enumeration protection on, you will no longer get a useful response to `fetchSignInMethodsWithEmail` and will need to do something different (or not enable email enumeration protection...) if that was important to you

**Reply by appsgenie** [reply]
got it. so if not using `fetchSignInMethodsForEmail` then should be safe to enabled enumeration protection? I am currently using a custom method to check if the email a user types to try to login with is existing in the database.

**Reply by mikehardy** [reply]
I cannot answer what is safe or not for you. I have stated what the impact is in my comment and you'll have to test + decide for your project's sake
